pytorch_lightning/loggers/neptune.py

```diff
@@ -523,6 +523,16 @@ def after_save_checkpoint(self, checkpoint_callback: "ReferenceType[ModelCheckpo
             file_names.add(model_name)
             self.experiment[f"{checkpoints_namespace}/{model_name}"].upload(key)
 
+        # log best model path and checkpoint
+        if checkpoint_callback.best_model_path:
```

Comment:
not an expert with neptune, but just wondering since we already upload all the best k models, shouldn't the best one already there and somehow if possible we can instead upload the best model checkpoint path/name?

Reply:
Internally Neptune has an asynchronous worker to consume operations (file upload, value assign, etc.) Essentially only the last file will be uploaded to the same namespace (it's like overriding which file should be uploaded).